packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -1444,9 +1444,11 @@ export class SnapController extends BaseController<
    */
   async clearState() {
     const snapIds = Object.keys(this.state.snaps);
-    snapIds.forEach((snapId) => {
-      this.#closeAllConnections(snapId);
-    });
+    if (this.#closeAllConnections) {
+      snapIds.forEach((snapId) => {
+        this.#closeAllConnections?.(snapId);
```

Comment:
Can this not be inferred?
```suggestion
        this.#closeAllConnections(snapId);
```

Reply:
Apparently not